### ISSUE TITLE
WOW stats not displaying when they should

### ISSUE THREAD

**Issue by kylebennison** [drewbennison/pelotonR#4]
Need to investigate.

Issue: Sometimes it says "need more data" when I'm certain i have rides from both this week and last week. Need to find the condition that causes it.

- Ruled out case when you are currently on a ride, thought that may be causing some NAs.

Condition that decides the message is 

```r
check_weeks <- week_sum_stats %>% group_by(week_num) %>% count()
                         
                         if(nrow(check_weeks) <2) {
```

So is there a case where you have data from this week and last week, but the nrow of check_weeks is < 2 for some reason? Maybe something with isoweek (Mon-Sun weeknumber)?

Maybe it's when you've worked out on Monday of this week, but last week you didn't work out on Monday, and our code filters out to only do a Monday to Monday comparison on Mondays, and Mon-Tues to Mon-Tues comparison on Tuesdays, etc. May be an unexpected behavior for users that is causing the confusion in these cases.

**Comment by kylebennison** [comment]
Code in question that may cause "Need More Data" when you've worked out on a Monday this week but didn't work out until Tuesday last week.

```r
# Summarise by week, then calculate differences
                         week_sum_stats <- joined_stats %>% 
                             filter(status == "COMPLETE") %>% 
                             filter(week_num  %in% c(isoweek(today()), isoweek(today()-7))) %>% 
                             filter(dplyr::between(as.Date(start_time), today() - days_since_monday, today()) | 
                                        dplyr::between(as.Date(start_time), today() - 7 - days_since_monday, today() - 7)) %>% 
                             group_by(week_num) 
```

**Comment by kylebennison** [comment]
That probably wouldn't be the case for dad since he always works out starting Monday and through Friday, so he shouldn't get that message on a Monday unless he's live riding the first workout of the week.

**Comment by kylebennison** [comment]
Solution for the above case could be to say something like 
"You're ahead of schedule this week: 
Total Output Last Week at this time: 0
Total Output This Week: 134"

**Comment by drewbennison** [comment]
Yes, the logic that I think it's using is what you said: if it's Monday of Week 2 and worked out today, but didn't work out until Wednesday last week, it will say not enough data. I believe I remember changing it to that to get rid of a different error lol.
